modules/integration/tests-integration/tests/src/test/java/org/wso2/carbon/integration/test/client/Wso2EventServer.java

```diff
@@ -52,34 +46,37 @@
 
 public class Wso2EventServer implements Runnable {
     private static Log log = LogFactory.getLog(Wso2EventServer.class);
+    private final String FILE_STREAM_DEFINITION_EXT = ".json";
+    AbstractStreamDefinitionStore streamDefinitionStore = new InMemoryStreamDefinitionStore();
     private ThriftDataReceiver thriftDataReceiver;
     private boolean eventReceived = false;
     private AtomicLong msgCount = new AtomicLong(0);
-    private final String FILE_STREAM_DEFINTIONS_EXT = ".json";
     private String testCaseResourceFolderName;
     private int listeningPort;
     private List<Event> preservedEventList = null;
     private boolean isPreservingEvents;
 
-    AbstractStreamDefinitionStore streamDefinitionStore = new InMemoryStreamDefinitionStore();
-
-
-    public Wso2EventServer(String testCaseResourceFolderName, int listeningPort, boolean isPreservingEvents){
+    public Wso2EventServer(String testCaseResourceFolderName, int listeningPort, boolean isPreservingEvents) {
         this.testCaseResourceFolderName = testCaseResourceFolderName;
         this.listeningPort = listeningPort;
         this.isPreservingEvents = isPreservingEvents;
     }
+
     public void startServer() throws DataBridgeException, StreamDefinitionStoreException {
-        msgCount.set(0);
-        start(listeningPort);
+        try {
+            msgCount.set(0);
+            start(listeningPort);
+        } catch (Throwable t) {
```

Comment:
@grainier I think I miscommunicated when we discussed this fix offline. Shall we put a try..catch..Throwable where the thread is run where the startServer() method is called. No need to catch it here.